src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -486,45 +486,48 @@ internal Task<List<RawFunctionMetadata>> SendFunctionMetadataRequest()
         }
 
         // parse metadata response into RawFunctionMetadata objects for WorkerFunctionMetadataProvider to further parse and validate
-        internal void ProcessFunctionMetadataResponses(FunctionMetadataResponses functionMetadataResponses)
+        internal void ProcessFunctionMetadataResponses(FunctionMetadataResponse functionMetadataResponse)
         {
             _workerChannelLogger.LogDebug("Received the worker function metadata response from worker {worker_id}", _workerId);
 
             var functions = new List<RawFunctionMetadata>();
 
-            foreach (var metadataResponse in functionMetadataResponses.FunctionLoadRequestsResults)
+            if (functionMetadataResponse.UseDefaultMetadataIndexing == false)
             {
-                var metadata = metadataResponse.Metadata;
-                if (metadata != null)
+                foreach (var metadata in functionMetadataResponse.FunctionMetadataResults)
                 {
-                    if (metadata.Status != null && metadata.Status.IsFailure(out Exception metadataRequestEx))
+                    if (metadata != null)
```

Comment:
nit: wonder if we can reverse some of these if statements to simplify, i.e. `if (metadata == null) break;` ?

Reply:
+1 - existing issue, but I think `continue` here if intent hasn't changed? I'm not 100% how this _should_ work - just commenting on the 1:1

Reply:
my bad, totally meant `continue` :D

Reply:
Done. Thanks.